nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -58,12 +62,26 @@ WaypointFollower::on_configure(const rclcpp_lifecycle::State & /*state*/)
   stop_on_failure_ = get_parameter("stop_on_failure").as_bool();
   loop_rate_ = get_parameter("loop_rate").as_int();
   waypoint_task_executor_id_ = get_parameter("waypoint_task_executor_plugin").as_string();
+  transform_tolerance_ = get_parameter("transform_tolerance").as_double();
+  global_frame_id_ = get_parameter("global_frame_id").as_string();
+  utm_frame_id_ = get_parameter("utm_frame_id").as_string();
+
+  global_frame_id_ = nav2_util::strip_leading_slash(global_frame_id_);
+  utm_frame_id_ = nav2_util::strip_leading_slash(utm_frame_id_);
 
   callback_group_ = create_callback_group(
     rclcpp::CallbackGroupType::MutuallyExclusive,
     false);
   callback_group_executor_.add_callback_group(callback_group_, get_node_base_interface());
 
+  std::vector<std::string> new_args = rclcpp::NodeOptions().arguments();
+  new_args.push_back("--ros-args");
+  new_args.push_back("-r");
+  new_args.push_back(std::string("__node:=") + this->get_name() + "_rclcpp_node");
+  new_args.push_back("--");
+  client_node_ = std::make_shared<rclcpp::Node>(
```

Comment:
Could you elaborate on this? the new node is not used in `nav_to_pose_client_` (which is an action client), but in the `from_ll_to_map_client_` (which is a service client). 

The implementation of the [`ServiceClient`wrapper](https://github.com/ros-planning/navigation2/blob/main/nav2_util/include/nav2_util/service_client.hpp#L40) in nav2_util seems to only accept a `Node` object which I don't know how to get inside the `on_configure` function. `shared_from_this()` in this case returns a `nav2_util::LifecycleNode` object, which the compiler cannot convert to type `Node`. 

On the other hand, the `nav_to_pose_client_` is created just as it was before (this part of the code is not changed in this PR).

Am I missing something?

Reply:
We do not allow for internal nodes to be created anymore, that's been long deprecated and we should be using executors for specific service-within-action calls that are blocking the current default executor's execution. We have already another executor with a setup callback group we use for another client we should be using here too. 

Optionally, you could update the service client to take in the new information to use instead of creating a new one **or** just give it the actual `this` node so that we create another internally to this in the constructor https://github.com/ros-planning/navigation2/blob/main/nav2_util/include/nav2_util/service_client.hpp#L43-L50. No need for an extra node. 

To take in the lifecycle node, just template the ServiceClient by the NodeT so that it can be used with multiple node types, including lifecycle. Example: https://github.com/ros-planning/navigation2/blob/cb90912074bacb250c2554b7897517ea776e3231/nav2_util/include/nav2_util/node_utils.hpp#L112-L124. You could also just update this to store only the NodeBaseInterface and have 2 constructors (normal node and lifecycle node) that grabs that for storage.

Reply:
Ok, understood. A [new callback group](https://github.com/ros-planning/navigation2/blob/main/nav2_util/include/nav2_util/service_client.hpp#L43) is created in the `ServiceClient` class though. Should that be removed and the [callback group used in the `nav_to_pose` client](https://github.com/ros-planning/navigation2/blob/534f01a851b7a6fcde61d6b3d7ffb53141cf2102/nav2_waypoint_follower/src/waypoint_follower.cpp#L72-L75) passed as an argument?

Reply:
This is fine. Its not 100% most ideal, but not a crazy issue. If you felt strongly, you could create a new constructor that takes in an executor / group to use instead of the node that we could use here to share it.

Reply:
I will resolve this since its not a big deal